sycl/unittests/kernel-and-program/Cache.cpp

```diff
@@ -124,38 +123,24 @@ static pi_result redefinedKernelRelease(pi_kernel kernel) { return PI_SUCCESS; }
 
 class KernelAndProgramCacheTest : public ::testing::Test {
 public:
-  KernelAndProgramCacheTest() : Plt{default_selector()} {}
+  KernelAndProgramCacheTest() : Mock{}, Plt{Mock.getPlatform()} {}
 
 protected:
   void SetUp() override {
-    if (Plt.is_host() || Plt.get_backend() != backend::opencl) {
-      std::clog << "This test is only supported on OpenCL devices\n";
-      std::clog << "Current platform is "
-                << Plt.get_info<info::platform::name>();
-      return;
-    }
-
-    Mock = std::make_unique<unittest::PiMock>(Plt);
-
-    setupDefaultMockAPIs(*Mock);
-    Mock->redefine<detail::PiApiKind::piclProgramCreateWithSource>(
+    Mock.redefine<detail::PiApiKind::piclProgramCreateWithSource>(
         redefinedProgramCreateWithSource);
-    Mock->redefine<detail::PiApiKind::piKernelGetInfo>(redefinedKernelGetInfo);
-    Mock->redefine<detail::PiApiKind::piKernelCreate>(redefinedKernelCreate);
-    Mock->redefine<detail::PiApiKind::piKernelRelease>(redefinedKernelRelease);
+    Mock.redefine<detail::PiApiKind::piKernelGetInfo>(redefinedKernelGetInfo);
+    Mock.redefine<detail::PiApiKind::piKernelCreate>(redefinedKernelCreate);
+    Mock.redefine<detail::PiApiKind::piKernelRelease>(redefinedKernelRelease);
```

Comment:
Do we need these still? Now that we have self-consistent PiMock, doesn't it default implementation provide suitable behavior already?

Reply:
They are not! In fact there's only one relevant redefinition here. The rest have been removed. Thanks!